grpc/federation/cel/plugin.go

```diff
@@ -204,13 +206,19 @@ func (i *CELPluginInstance) write(cmd []byte) error {
 	}()
 	select {
 	case err := <-i.instanceModErrCh:
+		// If the module instance is terminated,
+		// it is considered that the termination process has been completed.
+		i.closed = true
 		return err
 	case err := <-writeCh:
 		return err
 	}
 }
 
 func (i *CELPluginInstance) Close(ctx context.Context) error {
+	i.mu.Lock()
+	defer i.mu.Unlock()
```

Comment:
Currently, `write()` ( and `recvContent` ) is called from Call, Close, and ValidatePlugin. Since ValidatePlugin is guaranteed to be called sequentially, we attempted to control read and write access to `i.close` by acquiring a lock in `Call` and `Close` .

Reply:
Instead of using the lock to protect `i.closed`, I set the lock on methods that might be called concurrently.
( `Initialize` , `Call` , `Close` )